addons/stock/models/product.py

```diff
@@ -583,3 +583,22 @@ def _compute_total_route_ids(self):
             category = category.parent_id
             routes |= category.route_ids
         self.total_route_ids = routes
+
+
+class ProductUoM(models.Model):
+    _inherit = 'product.uom'
+
+    def write(self, values):
+        # Users can not update the factor if open stock moves are based on it
+        if 'factor' in values or 'factor_inv' in values:
+            stock_move_lines = self.env['stock.move.line'].search([
+                ('product_uom_id.category_id', '=', self.category_id.id),
+            ])
+
+            if stock_move_lines.filtered(lambda l: not l.location_id.should_bypass_reservation()):
```

Comment:
That would probably:
- not be efficient if large number of move lines
- cause access errors (UOM are available outside of stock)

Why not a simpler check such as https://github.com/odoo/odoo/blob/11.0/addons/stock/models/product.py#L511-L513 ? To me, your check is overkill. If you want to change the factor, then you probably should create a new UOM, otherwise you're going to screw up things.